WinUIGallery/Helper/NavigationHelper.cs

```diff
@@ -185,22 +187,20 @@ public RootFrameNavigationHelper(Frame rootFrame, Microsoft.UI.Xaml.Controls.Nav
             };
             this.CurrentNavView = currentNavView;
 
-            // must register back requested on navview
-            if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 6))
-            {
-                CurrentNavView.BackRequested += NavView_BackRequested;
-            }
+            CurrentNavView.BackRequested += NavView_BackRequested;
+            CurrentNavView.KeyDown += CurrentNavView_KeyDown;
+            CurrentNavView.KeyUp += CurrentNavView_KeyUp;
         }
 
-        private void NavView_BackRequested(Microsoft.UI.Xaml.Controls.NavigationView sender, Microsoft.UI.Xaml.Controls.NavigationViewBackRequestedEventArgs args)
+        private void NavView_BackRequested(NavigationView sender, NavigationViewBackRequestedEventArgs args)
         {
             TryGoBack();
         }
 
         private bool TryGoBack()
         {
-            // don't go back if the nav pane is overlayed
-            if (this.CurrentNavView.IsPaneOpen && (this.CurrentNavView.DisplayMode == Microsoft.UI.Xaml.Controls.NavigationViewDisplayMode.Compact || this.CurrentNavView.DisplayMode == Microsoft.UI.Xaml.Controls.NavigationViewDisplayMode.Minimal))
+            // Don't go back if the nav pane is overlayed.
+            if (this.CurrentNavView.IsPaneOpen && (this.CurrentNavView.DisplayMode == NavigationViewDisplayMode.Compact || this.CurrentNavView.DisplayMode == NavigationViewDisplayMode.Minimal))
             {
                 return false;
```

Comment:
bool navigated can be moved up, and this can simply return navigated.

Reply:
Fair point, updated.